conftest.py

```diff
@@ -0,0 +1,121 @@
+import json
+import os
+
+import boto3
+import pytest
+from moto import mock_iam, mock_s3
+
+
+@pytest.fixture(scope="function")
+def aws_credentials():
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+
+
+@pytest.fixture()
+def aws_test_user(aws_credentials):
+    with mock_iam():
+        user_name = "test-user"
+        policy_document = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": ["s3:ListBucket", "sqs:GetQueueUrl"],
+                    "Resource": "*",
+                },
+                {
+                    "Effect": "Deny",
+                    "Action": [
+                        "dynamodb:PutItem",
+                        "dynamodb:Scan",
+                        "s3:GetObject",
+                        "ses:SendRawEmail",
+                        "sqs:ReceiveMessage",
+                        "sqs:SendMessage",
+                        "ssm:GetParameter",
+                    ],
+                    "Resource": "*",
+                },
+            ],
+        }
+        client = boto3.client("iam", region_name="us-east-1")
+        client.create_user(UserName=user_name)
+        client.put_user_policy(
+            UserName=user_name,
+            PolicyName="policy1",
+            PolicyDocument=json.dumps(policy_document),
+        )
+        yield client.create_access_key(UserName="test-user")["AccessKey"]
+
+
+@pytest.fixture()
+def get_request_matching_file():
+    request_data = {
+        "queryStringParameters": {
+            "bucket": "ppod",
+            "file_type": "tar.gz",
+            "key_prefix": "upload",
+        },
+        "requestContext": {
+            "http": {
+                "method": "GET",
+            },
+        },
+    }
+    yield request_data
+
+
+@pytest.fixture()
+def get_request_no_files():
```

Comment:
In general, I would say having a fixture is really only useful if it gets used by multiple tests. For the `no_files` and `no_matching_file` cases, I think there's only one test that uses each of them right? So you can just create the dict (which will be much simpler for this lambda since there's no GET request) directly in the test function.

Reply:
Good point